examples/apps/cli/cli_uart.cpp

```diff
@@ -55,7 +55,7 @@
 #if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
 #define OPENTHREAD_CONFIG_CLI_UART_RX_BUFFER_SIZE 640
 #else
-#define OPENTHREAD_CONFIG_CLI_UART_RX_BUFFER_SIZE 512
```

Comment:
Prefer to keep it as is, but increase `OPENTHREAD_CONFIG_CLI_UART_RX_BUFFER_SIZE` to 640 in simulation and posix. 
Thoughts?